emma-language/src/main/scala/org/emmalanguage/compiler/Compiler.scala

```diff
@@ -65,6 +65,24 @@ trait Compiler extends AlphaEq with Source with Core with Backend {
     bld ++= transformations
     if (withPost)  bld ++= postProcess
     //@formatter:on
-    Function.chain(bld.result())
+    val steps = bld.result()
+
+    if (!printAllTrees) {
```

Comment:
I would agree with you, but unfortunately Scalastyle disagrees... (unless I write the whole `if else` thing on one line, but that hurts the readability).